pennylane/workflow/jacobian_products.py

```diff
@@ -741,10 +743,18 @@ def compute_vjp(self, tapes, dy):
                 raise NotImplementedError(
                     "Lightning device VJPs are not supported with jax jacobians."
                 )
-            vjp_f = self._device.vjp(
-                numpy_tape.measurements, dyi, **self._processed_gradient_kwargs
-            )
-            out = vjp_f(numpy_tape)
+
+            if self._uses_new_device:
+                if self._execution_config is not None:
+                    out = self._device.compute_vjp(numpy_tape, dyi, self._execution_config)
+                else:
+                    out = self._device.compute_vjp(numpy_tape, dyi)
+            else:  # pragma: no cover
```

Comment:
The `pragma: no cover` was adding to this line because the branch in question is used for old API devices (`lightning.kokkos` and `lightning.gpu`), which are not included in the testing suite.